sound/soc/sof/ipc4-pcm.c

```diff
@@ -1149,8 +1202,13 @@ static int sof_ipc4_pcm_pointer(struct snd_soc_component *component,
 
 	/* For delay calculation we need the host counter */
 	host_cnt = snd_sof_pcm_get_host_byte_counter(sdev, component, substream);
+
+	/* Store the original value to host_ptr */
 	host_ptr = host_cnt;
 
+	/* Scale value to DAI time in case DAI running at different rate */
+	host_cnt = sof_ipc4_time_host_to_dai(time_info, host_cnt);
```

Comment:
Why we don't scale the dai_cnt to host time-scale instead and save on back-forth scaling (here host to dai and later from dai to host)?

Reply:
Hmm, true, now that we can't assume which will wrap around faster, we might as well use host scale as the defaul and then we can avoid converting the resulting delay. Let me work on this. I have an additional fix in the works as well, will add it to this series.